cmd/ci-operator/main.go

```diff
@@ -1823,14 +1822,12 @@ func nodeNames(nodes []*api.StepNode) []string {
 	return names
 }
 
-func printDigraph(w io.Writer, steps []api.Step) error {
-	for _, step := range steps {
-		for _, other := range steps {
-			if step == other {
-				continue
-			}
-			if api.HasAnyLinks(step.Requires(), other.Creates()) {
-				if _, err := fmt.Fprintf(w, "%s %s\n", step.Name(), other.Name()); err != nil {
+func printDigraph(w io.Writer, steps api.OrderedStepList) error {
+	for i, step := range steps {
+		req := step.Step.Requires()
+		for _, other := range steps[:i] {
```

Comment:
Please add the following (or similar) comment here:
```golang
// steps have already been topologically sorted, 
// it means that if `a` depends on `b` then `b` will come before `a` on the list. 
// As a consequence, any dependencies of steps[i] will be found within steps[:i]
```

Reply:
Added.